pyNN/nest/standardmodels/electrodes.py

```diff
@@ -70,25 +81,51 @@ def _phase_correction(self, start, freq, phase):
         nest.SetStatus(self._device, {'phase': phase_fix})
 
     def _round_timestamp(self, value, resolution):
-        # subtraction by 1e-12 to match NEURON working
-        return round ((float(value)-1e-12)/ resolution) * resolution
+        return int(value/resolution+0.5) * resolution
+
+    def _check_step_times(self, times, amplitudes, resolution):
+        # ensure that all time stamps are non-negative
+        if not all(times>=0.0):
+            raise ValueError("Step current cannot accept negative timestamps.")
+        # ensure that times provided are of strictly increasing magnitudes
+        dt_times = numpy.diff(times)
+        if not all(dt_times>0.0):
+            raise ValueError("Step current timestamps should be monotonically increasing.")
+        # NEST specific: subtract min_delay from times (set to 0.0, if result is negative)
+        times = self._delay_correction(times)
+        # find the last element <= dt (we find >dt and then go one element back)
+        # this corresponds to the first timestamp that can be used by NEST for current injection
+        ctr = next((i for i,v in enumerate(times) if v > state.dt), len(times)) - 1
```

Comment:
I think this would be more efficient with numpy, e.g. `ctr = (v > state.dt).nonzero()[0][0]`

Reply:
Changed as suggested.

Reply:
Travis error - used wrong variable name, but persists after changing `v` to `times`. The expression evaluates to a boolean. Tried:
`ctr = [t for t in times if t>state.dt].nonzero()[0][0]` but this alters the indexing as we work on a sublist. Reverting for now.

Reply:
Found a neater way to do this:
`ctr = numpy.searchsorted(times, state.dt, side="right") - 1`